### PR TITLE
reqwest: use reqwest rather than raw hyper calls

### PR DESCRIPTION
**Pull request by mathstuf**
For now, the async method has been removed, but it will be added back.

### CODE REVIEW

**Comment by mgattozzi** [comment]
What issue is the async one being tracked on? Right now everything is sync so it doesn't really matter but I'd like to know.

**Comment by mathstuf** [comment]
None yet. Should implementation wait for reqwest to stabilize their API first?

**Comment by mgattozzi** [comment]
No it's fine. This is already not a fully stable library so I don't mind a littl bit of instability here. What matters to me is current users not realizing anything has really changed too much.

**Comment by mathstuf** [comment]
OK, I'll rebase this and open an issue for async support.

**Comment by mathstuf** [comment]
See #72.

**Comment by mgattozzi** [comment]
Awesome. Looks like there are still some conflicts for now (since I merged your other PR).

File: src/client.rs [hunk 1]
```diff
@@ -1,16 +1,11 @@
-// Tokio/Future Imports
-use futures::{ Future, Stream };
-use futures::future::ok;
-use tokio_core::reactor::Core;
-
 // Hyper Imports
-use hyper::{ Body, Headers, Uri, Method, Error };
-use hyper::client::{ Client, Request };
 use hyper::header::{ Authorization, Accept, ContentType,
```
Comment (mgattozzi) [inline src/client.rs]: Hmmmmm I'm wondering if later we should switch this over to the `http` crate

Comment (mathstuf) [reply src/client.rs]: Probably.

**Comment by mathstuf** [comment]
Conflicts fixed.

**Comment by mgattozzi** [comment]
Ah @mathstuf this was my bad I thought this was still pending when I had checked it and I looked again after merging some other changes that have caused conflicts. Would you be able to fix this PR up again? When you do I can merge it and we can add async later. I think I'll do another release after this gets in.

**Comment by mathstuf** [comment]
Updated. I also fixed some test compilation failures from #71. Two tests fail locally. Is there missing data?

**Comment by mgattozzi** [comment]
I don't think so. The tests require an auth token that's available on my machine. I really don't know the best way to set up a good testing infrastructure for this and also not hit some rate limiting problems. I'm going to have to contact Github support about setting up a testing token or something and look at how other APIs for it do it.

**Comment by mgattozzi** [comment]
I'm just going to run those tests locally and I'll merge if it's good!

**Comment by mathstuf** [comment]
The tests could inspect the generated `Request` in the structure rather than actually executing it.

**Comment by mgattozzi** [comment]
Yeah this would be fine for some things, but eventually I would like it to actually make a request and serialize to an expected data structure.

Anyways I'm getting a weird build conflict:

```
error: failed to select a version for `tokio-tls` (required by `hyper-tls`):
all possible versions conflict with previously selected versions of `tokio-tls`
  version 0.1.2 in use by tokio-tls v0.1.2
  possible versions to select: 0.1.3
```

I think this is an upstream issue with reqwest

**Comment by mathstuf** [comment]
Works here for me after a `cargo update`?

File: src/macros.rs [hunk 2]
```diff
@@ -22,11 +21,11 @@ macro_rules! from {
                 if f.request.is_ok() {
                     // We've checked that this works
                     let mut req = f.request.unwrap();
-                    let url = url_join(req.get_mut().uri(), $e)
+                    let url = req.url().join($e)
```
Comment (mgattozzi) [inline src/macros.rs]: I think the URL is not being set properly here:
```
URL: Ok(
    "https://api.github.com/repos"
)
```
The URL call for the tests should be "https://api.github.com/repos/mgattozzi/github-rs"
This is causing the tests to fail on my machine. This is where I had printed the line out.

Comment (mathstuf) [reply src/macros.rs]: D'oh. Url has some weird semantics with join. I've replaced it with parsing a formatted string instead.

**Comment by mgattozzi** [comment]
Awesome this works perfectly now! Thanks @mathstuf!

File: src/macros.rs (hunk 2 above)
Comment (mathstuf) [reply src/macros.rs]: Hrm, I see now that there are 3 places `join` is called. Will change them all in a new PR.

**Pull request merged**